REVIEW_COPY.md

```diff
@@ -0,0 +1,312 @@
+      -- it the same as a Move event.
+    | MoveOrStop
+
+type alias Coords =
+    { x : Float, y : Float }
+
+type alias Rect =
+    { x : Float, y : Float, width : Float, height : Float }
+```
+
+Now we can proof we are able to calculate the hour the cursor is over if we get `Msg`s that look like the above.
+
+```elm
+cursorAtHour : Msg -> Hour
+cursorAtHour { cursor, sliderPosition } =
+    let
+        dx =
+            cursor.x - sliderPosition.x
+
+        atMost =
+            min
+
+        atLeast =
+            max
+    in
+    (24 * (dx / sliderPosition.width))
+        |> floor
+        -- Ensure we get a number between 0 and 23,
+        -- even if the cursor moves to the left or right of the slider.
+        |> atMost 23
+        |> atLeast 0
+```
+
+All that's left to do is use this in our `update` function. When we get a `Start` event we update the `from` field in the model, and when we get a `MoveOrStop` event the `until` field.
+
+```elm
+update : Msg -> Model -> Model
+update msg model =
+    let
+        hour =
+            cursorAtHour msg
+    in
+    case msg.event of
+        Start ->
+            if coordsInRect msg.cursor msg.sliderPosition then
+                { from = hour
+                , until = hour + 1
+                }
+
+            else
+                model
+
+        MoveOrStop ->
+            { model
+                | until = 1 + max hour model.from
+            }
+
+coordsInRect : Coords -> Rect -> Bool
+coordsInRect =
+    Debug.todo "Implementation omitted for brevity."
+```
+
+And we have an app! Well almost, we've not discussed where these `Msg`s will be coming from. There's a bit of JavaScript responsible for that, which I'll talk more about in a bit. If you want to peak ahead check out the [time slider source code].
+
+## Second example, a polygon editor
+
+Our second example is a tool for editing polygons. We want users to be able to pick up a vertex of a polygon and move it somewhere else.
+
+![Dragging a vertex of a polygon][polygon.gif]
+
+You see those vertices? Those sure look like draggables! But these vertices are small and easy to miss, so we'd want our draggables to be bigger. We could achieve this by making the draggables invisible `div` elements centered on these vertices, but that gets us in trouble when the `div` elements are so close they overlap. At that point a click won't select the closest vertex but the top vertex.
+
+We're going to leave those draggables in the toolbox and see how far we can get without them. As usual we start by defining a model to store the polygon we're editing.
+
+```elm
+type alias Model =
+    { polygon : Polygon
+    , draggedVertex : Maybe Id
+    }
+
+type alias Polygon =
+    Array Coords
+
+type alias Id =
+    Int
+```
+
+We're using array indices as id's for the vertices of the polygon. Not a great solution in general, but it makes a bunch of update logic super simple allowing us to focus on the drag & drop aspects of the problem!
```

Comment:
would it make sense to say what you would prefer to do in real life instead? Just a couple words.

Reply:
Yeah, that would make sense. I took another approach: I changed the `Array` for a `Dict` which makes the existence of vertex id's explicit. I feel with that I can remove this paragraph entirely.